Services/CO.CDP.Organisation.WebApi/Api/Organisation.cs

```diff
@@ -157,26 +157,43 @@ public static RouteGroupBuilder UseOrganisationLookupEndpoints(this RouteGroupBu
                 operation.Responses["500"].Description = "Internal server error.";
                 return operation;
             });
+
         app.MapGet("/lookup",
-                async ([FromQuery] string name, IUseCase<string, Model.Organisation?> useCase) =>
-                await useCase.Execute(name)
-                    .AndThen(organisation => organisation != null ? Results.Ok(organisation) : Results.NotFound()))
-            .Produces<Model.Organisation>(StatusCodes.Status200OK, "application/json")
-            .Produces<ProblemDetails>(StatusCodes.Status401Unauthorized)
-            .Produces<ProblemDetails>(StatusCodes.Status404NotFound)
-            .Produces<ProblemDetails>(StatusCodes.Status500InternalServerError)
-            .WithOpenApi(operation =>
+            async ([FromQuery] string? name, [FromQuery] string? identifier, IUseCase<string, Model.Organisation?> useCase) =>
             {
-                operation.OperationId = "LookupOrganisation";
-                operation.Description = "Find an organisation.";
-                operation.Summary = "Find an organisation.";
-                operation.Tags = new List<OpenApiTag> { new() { Name = "Organisation - Lookup" } };
-                operation.Responses["200"].Description = "Organisations Associated.";
-                operation.Responses["401"].Description = "Valid authentication credentials are missing in the request.";
-                operation.Responses["404"].Description = "Organisation not found.";
-                operation.Responses["500"].Description = "Internal server error.";
-                return operation;
-            });
+                if (!string.IsNullOrEmpty(identifier))
+                {
+                    return await useCase.Execute($"identifier:{identifier}")
+                        .AndThen(organisation => organisation != null ? Results.Ok(organisation) : Results.NotFound());
```

Comment:
Rather than rely on a convention of the string, can we introduce a query object instead, please? 

You can create it in this controller, i.e.:

```csharp
useCase.Execute(new OrganisationQuery(name: name, identifier: identifier)
```

This way, the condition and validation will move to the use case. You'll need to throw an exception if both name and identifier are missing.

Reply:
@jakzal I have updated your suggestion in the code.